ray_lightning/ray_ddp.py

```diff
@@ -166,65 +150,88 @@ def train(self):
 
         self.global_to_local = self.get_local_ranks()
 
-        trainer = self.trainer
-        assert trainer is not None
-        trainer_ref = ray.put(trainer)
-        # Don't pickle self.trainer when training remotely.
-        self.trainer = None
+        model = self._model
+        model_ref = ray.put(model)
+        # Don't pickle the model when training remotely.
+        self._model = None
 
         queue = None
         if TUNE_INSTALLED and is_session_enabled():
             # Create communication queue and send to all the workers.
             queue = Queue(actor_options={"num_cpus": 0})
 
         futures = [
-            self.workers[i].execute.remote(self.train_remote, trainer_ref, i,
+            self.workers[i].execute.remote(self.train_remote, model_ref, i,
                                            queue)
             for i in range(self.num_workers)
         ]
 
         results = process_results(futures, queue)
         results, best_path, state_dict = results[0]
-        self.trainer = trainer
-        self.trainer.model.load_state_dict(state_dict)
-        if self.trainer.checkpoint_callback:
-            self.trainer.checkpoint_callback.best_model_path = best_path
+        self._results = results
```

Comment:
consider indicating whether certain things are required by the PTL interface or just used within ray_ddp

Reply:
Added a comment here. These are necessary for PTL.